jujugui/static/gui/src/app/views/state.js

```diff
@@ -368,9 +371,22 @@ YUI.add('juju-app-state', function(Y) {
       if (window.flags && window.flags.react) {
         if (Object.keys(queryValues).length > 0) {
           url = url.replace(/\/$/, '');
-          Object.keys(queryValues).forEach((key) => {
-            url += '?' + key + '=' + queryValues[key];
-          });
+          var keys = Object.keys(queryValues);
+          if (keys.length > 0) {
+            url += '?';
+            var i = 0;
+            keys.forEach((key) => {
```

Comment:
`forEach` passes the index as the second param which could be used instead of `i`